electron_app/electron-oidc.ts

```diff
@@ -179,6 +207,110 @@ function logout(
   );
 }
 
+async function silentRefresh(
+  authorityUrl: string,
+  config: IOidcConfig,
+  tokenObject: ITokenObject,
+  refreshCallback: Function,
+): Promise<void> {
+  const timeout = wait(tokenObject.expiresIn * 0.5 * 1000);
+  refreshTimeouts.set(authorityUrl, timeout);
+
+  await timeout;
+
+  if (!authoritiesToRefresh.includes(authorityUrl)) {
+    return;
+  }
+
+  // Build the Url Params from the Config.
+  const urlParams = {
+    client_id: config.clientId,
+    redirect_uri: config.redirectUri,
+    response_type: config.responseType,
+    scope: config.scope,
+    state: getRandomString(16),
+    nonce: getRandomString(16),
+    prompt: 'none',
+  };
+
+  const urlToLoad: string = `${authorityUrl}connect/authorize?${queryString.stringify(urlParams)}`;
+
+  // Open a new browser window and load the previously constructed url.
+  const authWindow = new BrowserWindow({show: false});
+
+  authWindow.loadURL(urlToLoad);
+
+  // Reject the Promise when the user closes the new window.
```

Comment:
Scheint mir ein Überbleibsel zu sein?

```suggestion
  // Throw an error, if the user closes the new window.
```